src/OmniSharp.Abstractions/FileWatching/IFileSystemWatcher.cs

```diff
@@ -9,8 +9,8 @@ public interface IFileSystemWatcher
         /// <summary>
         /// Call to watch a file or directory path for changes.
         /// </summary>
-        /// <param name="fileOrDirectoryPath">The file or directory path to watch.</param>
+        /// <param name="pathOrExtension">The file or directory path to watch.</param>
```

Comment:
Previously, you could. watch for files of any extension in a given directory. With this change, you can watch files of a given extension in any directory. However, there's still no way to watch files of a specific extension in a specific directory, and no way to watch all files in across all directories that omnisharp receives notifications about.

Maybe we should instead add an API that lets you subscribe to all notifications across all directories and let event subscribers handle filtering on extension/change type. Thoughts?

Reply:
there is a general need for globbing patterns support in OmniSharp - for example in case you define excluded paths in VS Code configuration, at the moment, OmniSharp doesn't know about that and still scans everything. I think what you suggest here could be developed together with a general globbing support in OmniSharp but personally I wouldn't hold up on this PR because it unblocks a very nice, and very welcome functionality that we've been missing for a while - watching for script and cake files

Reply:
@rchande @filipw I’m fine either way. I actually started out with a glob pattern matchet before @rchande pushed the improvements to FileSystemWatcher, but never really finishef since I never made friends with `Microsoft.Extensions.Filesystem.Globbing`. After discussion with @DustinCampbell in Slack we came tovthe conclusion that a simple `WatchExtension` would be sufficient to support wath we currently need for Scripting and Cake. I’ll let you decide on how to move forward and adjust this PR accordingly :smile:

Reply:
nvm the typos. It’s just me having a hard time texting on an iPhone..